src/js/components/alert.js

```diff
@@ -2,55 +2,57 @@
  * Copyright (C) 2018 The Trustees of Indiana University
  * SPDX-License-Identifier: BSD-3-Clause
  */
-import dispatchCustomEvent from '../utilities/dispatchCustomEvent';
-import { isNode } from '../utilities/domHelpers';
-
-export default class Alert {
-  constructor(element) {
-    // Instance properties
-    this.element = element;
-    this.closeAttribute = 'data-alert-close';
-    this.closeSelector = `[${this.closeAttribute}]`;
-    this.closeButton = null;
-
-    // bind methods
-    this._handleClick = this._handleClick.bind(this);
-
-    // Check to make sure that a DOM element was passed in for initialization
-    if (!isNode(this.element)) {
-      throw new TypeError(
-        'A DOM element should be passed as the first argument to initialize the alert'
-      );
-    }
-
-    this.init();
-  }
-
-  _handleClick() {
-    this.dismiss();
-  }
+import Component from './component';
 
-  dismiss() {
-    /**
-     * Custom event for handling removal of alerts from a page.
-     */
-    const dismissEvent = dispatchCustomEvent('alertDismiss', this.element, {
-      id: this.element.dataset.alert
-    });
-
-    if (!dismissEvent) return;
-
-    this.destroy();
-    this.element.remove();
+export default class Alert extends Component {
+  static get selector() {
+    return '[data-rvt-alert]';
   }
 
-  init() {
-    // Add click handlers
-    this.closeButton = this.element.querySelector(this.closeSelector);
-    this.closeButton.addEventListener('click', this._handleClick, false);
-  }
-
-  destroy() {
-    this.closeButton.addEventListener('click', this._handleClick, false);
+  static get methods() {
+    return {
+      init() {
+        console.log('Alert::init()');
+
+        this.closeAttribute = 'data-rvt-alert-close';
+        this.closeSelector = `[${this.closeAttribute}]`;
+        this.closeButton = null;
+
+        Component.bindMethodToDOMElement(this, 'dismiss', this.dismiss);
+        
+        this._handleClick = this._handleClick.bind(this);
+      },
+
+      connected() {
+        Component.dispatchComponentAddedEvent(this.element);
+
+        this.closeButton = this.element.querySelector(this.closeSelector);
+        this.closeButton.addEventListener('click', this._handleClick, false);
```

Comment:
We can take another pass at this, but click handlers like this should probably be moved to a native wicked-elements `onclick` method since it takes care of set up and tear down when the event listeners are added that way.

I think the only time we need to set up event listeners like this is when they are on something other that the component/element itself. So, say the `document` when we're listening for clicks outside the element itself.